File: Graphs - DFS/841. Keys and Rooms/solution.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited = [False]*len(rooms)
        visited[0] = True
        stack = [0]
        while stack:
            room = stack.pop()
            for key in rooms[room]:
                if not visited[key]:
                    visited[key] = True
                    stack.append(key)
        return all(visited)

File: Graphs - DFS/841. Keys and Rooms/solution.rs
impl Solution {
    pub fn can_visit_all_rooms(rooms: Vec<Vec<i32>>) -> bool {
        let mut visited = vec![false; rooms.len()];
        visited[0] = true;
        let mut stack = vec![0];
        while let Some(room) = stack.pop() {
            for &key in &rooms[room as usize] {
                if !visited[key as usize] {
                    visited[key as usize] = true;
                    stack.push(key);
                }
            }
        }
        visited.into_iter().all(|v| v)
    }
}